zerver/lib/rest.py

```diff
@@ -41,30 +41,19 @@ def _wrapped_view_func(request: HttpRequest, *args: object, **kwargs: object) ->
     return cast(ViewFuncT, _wrapped_view_func)  # https://github.com/python/mypy/issues/1927
 
 
-@default_never_cache_responses
-@csrf_exempt
-def rest_dispatch(request: HttpRequest, **kwargs: Any) -> HttpResponse:
-    """Dispatch to a REST API endpoint.
-
-    Unauthenticated endpoints should not use this, as authentication is verified
-    in the following ways:
-        * for paths beginning with /api, HTTP basic auth
-        * for paths beginning with /json (used by the web client), the session token
-
-    This calls the function named in kwargs[request.method], if that request
-    method is supported, and after wrapping that function to:
-
-        * protect against CSRF (if the user is already authenticated through
-          a Django session)
-        * authenticate via an API key (otherwise)
-        * coerce PUT/PATCH/DELETE into having POST-like semantics for
-          retrieving variables
-
-    Any keyword args that are *not* HTTP methods are passed through to the
-    target function.
-
-    Never make a urls.py pattern put user input into a variable called GET, POST,
-    etc, as that is where we route HTTP verbs to target functions.
+def get_target_view_function_or_response(
+    request: HttpRequest, kwargs: Dict[str, Any]
+) -> Union[Tuple[Callable[..., HttpResponse], Set[str]], HttpResponse]:
+    """
+    This pops handlers from kwargs and determines the target view function to be
+    used. When the method is not supported, it returns an HttpResponse object,
+    otherwise a tuple of the handler and a set of view flags are returned.
+
+    TODO: This is a transitional helper to remove handler delegation by HTTP
+    method from rest_dispatch. It decouples this routing functionality for the
+    ease of removal later. Eventually, we expect that paths with multiple
+    supported HTTP methods will be handled by class-based views provided by
+    Django.
```

Comment:
I rewrote this comment to document this function as itself. We have not made a decision to migrate to Django's class-based views, which do come with some downsides (extra indentation for all view code, ); 

The new comments include a proposal that we remove the mutation of the `kwargs` parameter, since I think it's not actually used as best I can tell on a quick audit, and is a bad pattern.

Reply:
> I rewrote this comment to document this function as itself. We have not made a decision to migrate to Django's class-based views, which do come with some downsides (extra indentation for all view code, );

I see. Will start a discussion on CZO on this topic.